client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -1508,6 +1511,31 @@ private TxnSegments convert(CreateTxnResponse response) {
                 });
     }
 
+    @Override
+    public AsyncIterator<io.pravega.controller.stream.api.grpc.v1.Controller.TxnResponse> listCompletedTransactions(Stream stream) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        Preconditions.checkNotNull(stream, "stream");
+        final long requestId = requestIdGenerator.get();
+        long traceId = LoggerHelpers.traceEnter(log, "listCompletedTransactions", stream, requestId);
+
+        try {
+            final Function<ContinuationToken, CompletableFuture<Map.Entry<ContinuationToken, Collection<io.pravega.controller.stream.api.grpc.v1.Controller.TxnResponse>>>> function =
+                    token -> this.retryConfig.runAsync(() -> {
+                        RPCAsyncCallback<ListCompletedTxnResponse> callback = new RPCAsyncCallback<>(traceId, "listCompletedTransactions", stream);
+
+                        new ControllerClientTagger(client, timeoutMillis).withTag(requestId, LIST_COMPLETED_TRANSACTIONS, stream.getScope(), stream.getStreamName())
+                                .listCompletedTransactions(ListCompletedTxnRequest.newBuilder()
+                                        .setStreamInfo(ModelHelper.createStreamInfo(stream.getScope(), stream.getStreamName()))
+                                        .setContinuationToken(token)
+                                        .build(), callback);
+                        return callback.getFuture().thenApplyAsync(listCompletedTxnResponse -> new AbstractMap.SimpleEntry<>(listCompletedTxnResponse.getContinuationToken(), listCompletedTxnResponse.getResponseList()), this.executor);
+                    }, this.executor);
+            return new ContinuationTokenAsyncIterator<>(function, ContinuationToken.newBuilder().build());
+        } finally {
+            LoggerHelpers.traceLeave(log, "listCompletedTransactions", traceId);
```

Comment:
`LIST_COMPLETED_TRANSACTIONS` should be used

Reply:
Done as suggested.